### PR TITLE
Write test for dropdown component

### PR DESCRIPTION
**Pull request by anishpawaskar**
## Issue Ticket Number
Ticket: https://github.com/Real-Dev-Squad/members-site/issues/104

## Description
Write test for dropdown component
- write test for `Dropdown``
- write test for `DropdownPresentation` 
- write test for `UserLogout` component

**Breaking Changes**

-   [ ] Yes
-   [x] No

**Development Tested?**

-   [x] Yes
-   [ ] No

**Under Feature Flag**

-   [ ] Yes
-   [x] No

**Database Changes**

-   [ ] Yes
-   [x] No

## Test Coverage

image

### CODE REVIEW

File: src/test/unit/components/dropdown/component/UserLogout.test.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    const signoutButton = screen.getByRole("button", { name: "Sign out" });
+    expect(signoutButton).toBeInTheDocument();
+    fireEvent.click(signoutButton);
+
+    await waitFor(() => {
+      expect(mockLogoutUser).toHaveBeenCalled();
+    });
+  });
+
+  test("should logout fail", async () => {
```
Comment (yesyash) [inline src/test/unit/components/dropdown/component/UserLogout.test.tsx:32]:
```suggestion
  test("should fail logout", async () => {
```

File: src/components/Dropdown/components/UserLogout.tsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import { useDispatch } from "react-redux";
-import { useLogoutUserMutation } from "@/src/services/logoutApi";
+import { useLogoutUserMutation } from "../../../services/logoutApi";
 
-import { setIsLoggedIn, setUserData } from "@/src/store/global";
+import { setIsLoggedIn, setUserData } from "../../../store/global";
 
-import { notifyError, notifySuccess } from "@/src/utils/toast";
+import { notifyError, notifySuccess } from "../../../utils/toast";
```
[the remark this answers is not in the document]
Comment (anishpawaskar) [reply src/components/Dropdown/components/UserLogout.tsx:6]: Currently jest is not setup properly in member site so if try to run test and the component im test have any absolute import then all test cases written for that component fails

**Comment by ajoykumardas12** [comment]
> write test for DropdownPresentation
This is not present (not using anymore I think).

**Comment by anishpawaskar** [comment]
> > write test for DropdownPresentation
> 
> This is not present (not using anymore I think).

Yes after discussing with vinayak i have deprecated that component

File: src/test/unit/components/dropdown/Dropdown.test.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    expect(taskLink).toBeInTheDocument();
+
+    const identityLink = screen.getByRole("link", { name: "Identity" });
+    expect(identityLink).toBeInTheDocument();
+
+    const signoutButton = screen.getByRole("button", { name: "Sign out" });
+    expect(signoutButton).toBeInTheDocument();
+  });
+
+  test("should navigate to a different route when a link is pressed", () => {
```
Comment (ajoykumardas12) [inline src/test/unit/components/dropdown/Dropdown.test.tsx:40]: Not seeing any navigation to different route in tests.

File: src/test/unit/components/dropdown/Dropdown.test.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    expect(profileLink).toBeInTheDocument();
+
+    const taskLink = screen.getByRole("link", { name: "Tasks" });
+    expect(taskLink).toBeInTheDocument();
+
+    const identityLink = screen.getByRole("link", { name: "Identity" });
+    expect(identityLink).toBeInTheDocument();
+
+    const signoutButton = screen.getByRole("button", { name: "Sign out" });
+    expect(signoutButton).toBeInTheDocument();
```
Comment (ajoykumardas12) [inline src/test/unit/components/dropdown/Dropdown.test.tsx:37]: Do you think declaring all variable first, then testing for being in document would be more readable?

File: src/test/unit/components/dropdown/component/UserLogout.test.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import { fireEvent, screen, waitFor } from "@testing-library/react";
+import { useLogoutUserMutation } from "../../../../../services/logoutApi";
+import { renderWithProviders } from "../../../../../test__utils/renderWithProvides";
+
+import { UserLogout } from "../../../../../components/Dropdown/components/UserLogout";
+
+jest.mock("../../../../../services/logoutApi", () => ({
+  useLogoutUserMutation: jest.fn(),
+}));
```
[the remark this answers is not in the document]
Comment (vinit717) [reply src/test/unit/components/dropdown/component/UserLogout.test.tsx:9]: We can use to mock the API without dependent on backend

[the remark this answers is not in the document]
Comment (anishpawaskar) [reply src/test/unit/components/dropdown/component/UserLogout.test.tsx:9]: yes i know that

File: src/test/unit/components/dropdown/Dropdown.test.tsx (hunk 3 above)
Comment (anishpawaskar) [reply src/test/unit/components/dropdown/Dropdown.test.tsx:40]: it is getting changed in #121

**Review by ajoykumardas12 — APPROVED** [review]

**Review by satyam73 — APPROVED** [review]
LGTM

**Pull request merged**